File: c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"

bool is_armstrong_number(int candidate) {
    int power = 0;
    for (int i = candidate; i > 0; i /= 10) ++power;

    int sum = 0;
    for (int i = candidate; i > 0; i /= 10) {
        int digit = i % 10;
        int raised = 1;
        for (int i = power; i > 0; --i) raised *= digit;
        sum += raised;
    }
    return candidate == sum;
}

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        for (int i = numberToCheck, pow = (i + "").length(); i > 0; i /= 10) {
            final int digit = i % 10;
            int power = 1;
            for (int j = 0; j < pow; j++) {
                power *= digit;
            }
            numberToCheck -= power;
        }
        return numberToCheck == 0;
    }

}
